app/domain/SlpFileWriter.js

```diff
@@ -37,6 +46,71 @@ export default class SlpFileWriter {
 
   updateSettings(settings) {
     this.folderPath = settings.targetFolder;
+    this.obsIP = settings.obsIP;
+    this.obsSourceName = settings.obsSourceName;
+    this.obsPassword = settings.obsPassword;
+  }
+
+  async connectOBS() {
+    if (this.obsIP && this.obsSourceName) {
+      // if you send a password when authentication is disabled, OBS will still connect
+      await this.obs.connect({address: this.obsIP, password: this.obsPassword});
+      const obsScenes = await this.obs.send("GetSceneList");
+      this.scenes = obsScenes.scenes;
+    }
+  }
+
+  disconnectOBS() {
+    this.obs.disconnect();
+  }
+
+  setStatus(value) {
+    this.statusOutput.status = value;
+    console.log(`Status changed: ${value}`);
+    const scenes = this.scenes || [];
+    const pairs = _.flatMap(scenes, (scene) => {
+      const sources = scene.sources || [];
+      return _.map(sources, (source) => ({scene: scene.name, source: source.name}));
+    });
+    _.forEach(pairs, (pair) => {
+      if (pair.source !== this.obsSourceName) {
+        return;
+      }
+
+      const res = this.obs.send("SetSceneItemProperties", 
+        {"scene-name": pair.scene, "item": this.obsSourceName, "visible": value});
+      console.log(res);
+    });
+  }
+
+  handleStatusOutput() {
+    const setTimer = () => {
+      if (this.statusOutput.timeout) {
+        // If we have a timeout, clear it
+        clearTimeout(this.statusOutput.timeout);
+      }
+
+      this.statusOutput.timeout = setTimeout(() => {
+        // If we timeout, set and set status
+        this.setStatus(false);
+      }, 100);
+    }
+
+    if (this.statusOutput.status) {
+      // If game is currently active, reset the timer
+      setTimer();
+      return;
+    }
+    if (this.currentFile.metadata.lastFrame < -70) {
```

Comment:
maybe this check should be before the previous if block? There's no reason to set the timer if we aren't setting the status to true right?

Reply:
agreed, I moved them around